wordpress-mobile/WordPress-iOS
# Refactor exportable media protocol

This PR refactor the `MediaExporter` protocol to make all the exporters to have a common interface to invoke them. 

This converts all the exporters to have a common return type on the callback and then allows them to be invoked using the same code and the answer to be analysed with a common code.

This is a preliminary step to allow common interface on all exporter to report progress.

To test:
 - Check in the app the following features:
 - Media Library
 - Gravatar selection
 - Site Icon selection
 - Aztec editor, insert images
